### PR TITLE
IGNITE-17818 Optimize RocksDB-based sorted index scan

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-17818

### CODE REVIEW

File: modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/index/impl/TestSortedIndexStorage.java [hunk 1]
```diff
@@ -118,21 +114,31 @@ public Cursor<IndexRow> scan(
...
+        if (lowerBound == null && upperBound == null) {
+            data = index;
+        } else if (lowerBound == null) {
+            data = index.headMap(upperBound.byteBuffer());
+        } else if (upperBound == null) {
+            data = index.tailMap(lowerBound.byteBuffer());
+        } else {
+            try {
+                data = index.subMap(lowerBound.byteBuffer(), upperBound.byteBuffer());
+            } catch (IllegalArgumentException e) {
```
Comment (ibessonov) [inline modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/index/impl/TestSortedIndexStorage.java:136]: This is new for me, how can it happen?

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/sorted/PageMemorySortedIndexStorage.java [hunk 2]
```diff
@@ -120,26 +119,60 @@ public void remove(IndexRow row) {
...
         try {
-            cursor = sortedIndexTree.find(
-                    toSortedIndexRowKey(lowerBound),
-                    toSortedIndexRowKey(upperBound),
-                    (flags & GREATER_OR_EQUAL) != 0,
-                    (flags & LESS_OR_EQUAL) != 0,
-                    null,
-                    null
-            );
+            Cursor<SortedIndexRow> cursor = sortedIndexTree.find(lower, upper, includeLower, includeUpper, null, null);
```
Comment (ibessonov) [inline modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/sorted/PageMemorySortedIndexStorage.java:130]: Passing these flags is pointless I believe. Simple `sortedIndexTree.find(lower, upper)` will give us the same result

File: modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/index/impl/TestSortedIndexStorage.java (hunk 1 above)
Comment (sashapolo) [reply modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/index/impl/TestSortedIndexStorage.java:136]: This exception is thrown if the lowerBound is larger than the upperBound

Comment (ibessonov) [reply modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/index/impl/TestSortedIndexStorage.java:136]: Ok, thank you!

**Review by ibessonov — APPROVED** [review]

**Pull request merged**